### PR TITLE
test-lib: add binary availability in Dockerfile check

### PR DESCRIPTION
**Pull request by pkubatrh**
Required-by: s2i-nodejs-container#230

### CODE REVIEW

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
rhel CI failing because of:
```
stderr: fatal: unable to connect to github.com:
github.com[0: 192.30.253.113]: errno=Connection timed out
```

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
nodejs PR needed rebasing for centos fixes, one more [test]

**Comment by pkubatrh** [comment]
Made the test more generic so we can now check any binary not only npm.

[test]

**Comment by pkubatrh** [comment]
[test]

File: test-lib.sh [hunk 1]
```diff
@@ -285,6 +285,32 @@ function ct_npm_works() {
...
+  tmpdir=$(mktemp -d)
+  : "  Testing $binary in build from Dockerfile"
+
+  # Create Dockerfile that looks for the binary
+  cat <<EOF >"$tmpdir/Dockerfile"
+FROM $IMAGE_NAME
+RUN which $binary
+EOF
+  # Build an image, looking for expected path in the output
+  if ! docker build -f "$tmpdir/Dockerfile" --no-cache "$tmpdir" 2>&1 | grep "$binary_path"; then
```
Comment (phracek) [inline test-lib.sh:307]: Is it enough to check only `/opt/rh`? What about to add as a parameter also `version` and check `full path` like was before in `grep -E`. To be more precise.

Comment (pkubatrh) [reply test-lib.sh:307]: You can already do that via the second argument, `/opt/rh` is only the default.

Comment (phracek) [reply test-lib.sh:307]: What about `grep "^$binary_path"`?

Comment (phracek) [reply test-lib.sh:307]: I am still thinking if it makes sense to somehow automatically "calculate" full binary path for a binary but the effort seems to be really high then expectation. Let's leave it to `^$binary_path'.

Comment (pkubatrh) [reply test-lib.sh:307]: I would leave that up to the user of the function. I have added the `^` to the default value though. Thanks for the idea!

**Comment by pkubatrh** [comment]
[test]

File: test-lib.sh (hunk 1 above)
Comment (phracek) [reply test-lib.sh:307]: Thanks. LGTM.

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
Thanks for review, tests are green now so merging

**Pull request merged**